drivers/led/is31fl3733.c

```diff
@@ -0,0 +1,298 @@
+{
+	return is31fl3733_led_set_brightness(dev, led, IS31FL3733_MAX_BRIGHTNESS);
+}
+
+static int is31fl3733_led_off(const struct device *dev, uint32_t led)
+{
+	return is31fl3733_led_set_brightness(dev, led, 0);
+}
+
+static int is31fl3733_led_write_channels(const struct device *dev, uint32_t start_channel,
+					 uint32_t num_channels, const uint8_t *buf)
+{
+	const struct is31fl3733_config *config = dev->config;
+	struct is31fl3733_data *data = dev->data;
+	int ret = 0U;
+	uint8_t *pwm_start;
+
+	if ((start_channel + num_channels) > IS31FL3733_MAX_LED) {
+		return -EINVAL;
+	}
+	pwm_start = data->scratch_buf + start_channel;
+	/* Set PWM and LED target registers as first byte of each transfer */
+	*pwm_start = start_channel;
+	memcpy((pwm_start + 1), buf, num_channels);
+
+	/* Write LED PWM states */
+	ret = is31fl3733_select_page(dev, CMD_SEL_PWM);
+	if (ret < 0) {
+		return ret;
+	}
+	LOG_HEXDUMP_DBG(pwm_start, (num_channels + 1), "PWM states");
+	return i2c_write_dt(&config->bus, pwm_start, num_channels + 1);
+}
+
+static int is31fl3733_init(const struct device *dev)
+{
+	const struct is31fl3733_config *config = dev->config;
+	struct is31fl3733_data *data = dev->data;
+	int ret = 0U;
+	uint8_t dummy;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
+	if ((config->sdb.port != NULL) && !gpio_is_ready_dt(&config->sdb)) {
+		LOG_ERR("GPIO SDB pin not ready");
+		return -ENODEV;
+	}
+
+	ret = is31fl3733_select_page(dev, CMD_SEL_FUNC);
+	if (ret < 0) {
+		return ret;
+	}
+	/*
+	 * read reset reg to reset all registers to POR state,
+	 * in case we are booting from a warm reset.
+	 */
+	ret = i2c_reg_read_byte_dt(&config->bus, RESET_REG, &dummy);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (config->sdb.port != NULL) {
+		/* Set SDB pin high to exit hardware shutdown */
+		ret = gpio_pin_configure_dt(&config->sdb, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	/* Select function page after LED controller reset */
+	ret = is31fl3733_select_page(dev, CMD_SEL_FUNC);
+	if (ret < 0) {
+		return ret;
+	}
+	/* Set global current control register based off devicetree value */
+	ret = i2c_reg_write_byte_dt(&config->bus, GLOBAL_CURRENT_CTRL_REG,
+				config->current_limit);
+	if (ret < 0) {
+		return ret;
+	}
+	/* As a final step, we exit software shutdown, disabling display
+	 * blanking. We also set the LED controller sync mode here.
+	 */
+	data->conf_reg = (config->sync << CONF_REG_SYNC_SHIFT) | CONF_REG_SSD_MASK;
+	ret = i2c_reg_write_byte_dt(&config->bus, CONF_REG, data->conf_reg);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Enable all LEDs. We only control LED brightness in this driver. */
+	data->scratch_buf[0] = 0x0;
+	memset(data->scratch_buf + 1, 0xFF, (IS31FL3733_MAX_LED / 8));
+	ret = is31fl3733_select_page(dev, CMD_SEL_LED);
+	if (ret < 0) {
+		return ret;
+	}
+	return i2c_write_dt(&config->bus, data->scratch_buf,
```

Comment:
Could you please add nl before the return statements in the entire driver?

Reply:
I have added newlines before return statements at the end of each driver function (assuming that is what you mean)